numismatic/feeds/poloniex.py

```diff
@@ -33,110 +40,118 @@ class PoloniexWebsocketClient(WebsocketClient):
-                PoloniexWebsocketClient._handle_trade(msg, subscription)
+            if msg_type == 'i':
+                PoloniexWebsocketClient._handle_orderbook(seq, data[1], subscription)
+            elif msg_type == 'o':
+                PoloniexWebsocketClient._handle_orderbook_removemodify(seq, data, subscription)
+            elif msg_type == 't':
+                PoloniexWebsocketClient._handle_trade(seq, data, subscription)
 
-            return STOP_HANDLERS
+        return STOP_HANDLERS
 
     @staticmethod
-    def _handle_order_book(msg, subscription):
+    def _handle_orderbook(seq, market_info, subscription):
         '''
             Poloniex order book, this will provide a snapshot
             of the order book
             TODO Put format here
         '''
-
-        if msg[2][0][0] == 'i':
-            raw_data = msg[2][0][1]
-
-            for key, value in raw_data.items():
-                if key == 'orderBook':
-                    for ask_price, volume in value[0].items():
-                        event = Order(
-                            exchange=subscription.exchange,
-                            symbol=subscription.symbol,
-                            price=ask_price,
-                            volume=volume,
-                            type='ASK'
-                        )
-                        subscription.event_stream.emit(event)
-                    for bid_price, volume in value[1].items():
-                        event = Order(
-                            exchange=subscription.exchange,
-                            symbol=subscription.symbol,
-                            price=bid_price,
-                            volume=volume,
-                            type='BID'
-                        )
-                        subscription.event_stream.emit(event)
+        for key, value in market_info.items():
+            if key == 'orderBook':
+                for ask_price, volume in value[0].items():
+                    event = Order(
+                        exchange=subscription.exchange,
+                        symbol=subscription.symbol,
+                        price=ask_price,
+                        volume=volume,
+                        type='ASK',
+                        sequence=seq,
+                    )
+                    subscription.event_stream.emit(event)
+
+                for bid_price, volume in value[1].items():
+                    event = Order(
+                        exchange=subscription.exchange,
+                        symbol=subscription.symbol,
+                        price=bid_price,
+                        volume=volume,
+                        type='BID',
+                        sequence=seq,
+                    )
+                    subscription.event_stream.emit(event)
 
     @staticmethod
-    def _handle_trade(msg, subscription):
+    def _handle_trade(seq, data, subscription):
         '''
             Poloniex trade format:
```

Comment:
Having these example messages here is very nice. We should put that in the code for all the other exchanges as well.